packages/aserto/test/test_directory.py

```diff
@@ -0,0 +1,149 @@
+    Relation,
+)
+
+
+@dataclass(frozen=True)
+class TestData:
+    client: Directory
+    obj_1: Object
+    obj_2: Object
+    obj_3: Object
+    relation_1: Relation
+    relation_2: Relation
+
+
+@pytest.fixture(scope="module")
+def directory_client(topaz):
+    client = Directory(address=topaz.directory.address, ca_cert=topaz.directory.ca_cert_path)
+
+    yield client
+
+    client.close()
+
+
+@pytest.fixture
+def directory(directory_client):
+    key_1 = uuid.uuid4().hex
+    key_2 = uuid.uuid4().hex
+
+    obj_1 = directory_client.set_object(Object(key=key_1, type="user", display_name="test user"))
+    obj_2 = directory_client.set_object(Object(key=key_2, type="group", display_name="test group"))
+    obj_3 = directory_client.set_object(
+        Object(key=key_2, type="user", display_name="another test user")
+    )
+
+    relation_1 = directory_client.set_relation(
+        relation={
+            "subject": {"key": obj_1.key, "type": obj_1.type},
+            "object": {"key": obj_2.key, "type": obj_2.type},
+            "relation": "member",
+        }
+    )
+
+    relation_2 = directory_client.set_relation(
+        relation={
+            "subject": {"key": obj_1.key, "type": obj_1.type},
+            "object": {"key": obj_3.key, "type": obj_3.type},
+            "relation": "manager",
+        }
+    )
+
+    yield TestData(
+        client=directory_client,
+        obj_1=obj_1,
+        obj_2=obj_2,
+        obj_3=obj_3,
+        relation_1=relation_1,
+        relation_2=relation_2,
+    )
+
+    relations = directory_client.get_relations(page=PaginationRequest(size=30)).results
+    for rel in relations:
+        directory_client.delete_relation(
+            subject_type=rel.subject.type,
+            subject_key=rel.subject.type,
+            object_type=rel.object.type,
+            object_key=rel.object.key,
+            relation_type=rel.relation,
+        )
+
+    objects = directory_client.get_objects(page=PaginationRequest(size=30)).results
+    for obj in objects:
+        directory_client.delete_object(key=obj.key, type=obj.type)
+
+
+def test_delete_object(directory):
+    obj_to_delete = directory.client.get_object(key=directory.obj_1.key, type=directory.obj_1.type)
+    directory.client.delete_object(key=obj_to_delete.key, type=obj_to_delete.type)
+    with pytest.raises(NotFoundError):
+        directory.client.get_object(key=directory.obj_1.key, type=directory.obj_1.type)
+
+
+def test_get_object(directory):
+    obj = directory.client.get_object(key=directory.obj_1.key, type=directory.obj_1.type)
+    assert obj.key == directory.obj_1.key
+    assert obj.type == directory.obj_1.type
+    assert obj.display_name == directory.obj_1.display_name
+
+
+def test_object_not_found(directory):
+    key = uuid.uuid4().hex
+    with pytest.raises(NotFoundError):
+        directory.client.get_object(key=key, type="user")
+
+
+def test_object_invalid_arg(directory):
+    with pytest.raises(TypeError):
```

Comment:
This isn't quite testing what we need. You're getting a TypeError because you're calling a function that takes 2 parameters with no default values, but are only passing one.
A better test would be to pass an empty string as the type parameter (i.e. `type=""`).